### DISCUSSION TITLE
KAM average

### DISCUSSION THREAD

**Discussion by arthurribe** [mtex-toolbox/mtex#819] [Ask Anything] [answered]
Hello MTEX comunity,

I'm trying to find a mean value for the KAM in a recrystallized nickel-based superalloy. I'm using the following code:

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));
% remove one-three pixel grains
ebsd(grains(grains.grainSize <= 3)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));

grains = smooth(grains,5);

plot(ebsd('indexed'),ebsd('indexed').orientations)
hold on
plot(grains.boundary,'lineWidth',1.5)
hold off

ebsd = ebsd.gridify;

kam = ebsd.KAM / degree;

% lets plot it
plot(ebsd,kam,'micronbar','off')
caxis([0,5])
mtexColorbar
mtexColorMap LaboTeX

plot(ebsd,ebsd.KAM('threshold',2.5*degree,'order',5) ./ degree,'micronbar','off')
caxis([0,2])
mtexColorbar
mtexColorMap LaboTeX
hold on
plot(grains.boundary,'lineWidth',1.5)
hold off

a = mean(kam, 'all');

However, matlab gives the following error:

Error using sum
Invalid option. Option must be 'double', 'native', 'default', 'omitnan' or 'includenan'.

Error in mean (line 116)
        y = sum(x, dim, flag) ./ size(x,dim);
Can anybody help me?

**Comment by kilir** [comment, marked as answer]
Hi,

since the kam often contains nan, try nanmean().

Cheers,
Rüdiger

**Reply by arthurribe** [reply]
Hi Rudiger,
Thanks for the answer, but it now gives the error.
Error using size
Dimension argument must be a positive integer scalar within indexing range.

Error in nanmean (line 43)
count = size(x,dim) - sum(nans,dim);
Which is strange since I'm not using "size" in the code.

**Reply by kilir** [reply]
It looks like you are using the nanmean() provided by the Matlab Statistics and ML Toolbox, I can't reproduce your observation with the `nanmean()` from Mtex. If you use a recent version of Matlab `mean(kam(:),'omitnan')` should also work. If not, providing more info might be helpful (what tpye is the variable kam reported as, mtex and matlab version, maybe attach data etc.).

Rüdiger

**Reply by arthurribe** [reply]
Thanks a lot, Rüdiger ! It now worked with this code "mean(kam(:),'omitnan')".

Best Regards!
Arthur
